### PR TITLE
Feat/cosigned/add support for webhook name flag

### PR DESCRIPTION
**Pull request by vpnachev**
#### Summary
With https://github.com/sigstore/cosign/pull/1726, cosigned will support new flag `--webhook-name` that controls the name of the mutating and validating webhook configurations. With this PR, this flag is also configurable via the chart. 

#### Ticket Link

Fixes

#### Release Note

```release-note
The cosigned flag `--webhook-name` can be now set via `.Values.cosign.webhookName` field, which is required with default value `cosigned.sigstore.dev`.
```

/cc @hectorj2f

### CODE REVIEW

File: charts/cosigned/README.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ The following table lists the configurable parameters of the cosigned chart and
 | commonTolerations | list | `[]` |  |
 | cosign.cosignPub | string | `""` |  |
 | cosign.secretKeyRef.name | string | `""` |  |
+| cosign.webhookName | string | `""` |  |
```
Comment (hectorj2f) [inline charts/cosigned/README.md:18]: This should reflect the default value `cosigned.sigstore.dev`

File: charts/cosigned/templates/_helpers.tpl [hunk 2]
```diff
@@ -116,6 +116,17 @@ Create the image path for the passed in image field
...
 {{- end -}}
 
+{{/*
+Cosigned webhook name
+*/}}
+{{- define "cosigned.webhookName" -}}
+{{- if .Values.cosign.webhookName -}}
+{{- .Values.cosign.webhookName -}}
+{{- else -}}
+cosigned.sigstore.dev
```
Comment (hectorj2f) [inline charts/cosigned/templates/_helpers.tpl:126]: the values.yaml should set the default value, so you don't set it here in the helpers.

File: charts/cosigned/values.yaml [hunk 3]
```diff
@@ -3,6 +3,7 @@ cosign:
     name: ""
   # add the values in base64 encoded
   cosignPub: ""
+  webhookName: ""
```
Comment (hectorj2f) [inline charts/cosigned/values.yaml:6]: set here `cosigned.sigstore.dev`

File: charts/cosigned/README.md (hunk 1 above)
Comment (vpnachev) [reply charts/cosigned/README.md]: Done

File: charts/cosigned/templates/_helpers.tpl (hunk 2 above)
Comment (vpnachev) [reply charts/cosigned/templates/_helpers.tpl]: Done

File: charts/cosigned/values.yaml (hunk 3 above)
Comment (vpnachev) [reply charts/cosigned/values.yaml]: Done

File: charts/cosigned/templates/webhook/deployment_webhook.yaml [hunk 4]
```diff
@@ -48,6 +48,9 @@ spec:
 {{- end }}
 {{- end }}
         args:
+        {{- if semverCompare ">= 1.8-0" .Chart.AppVersion }}
```
Comment (vpnachev) [inline charts/cosigned/templates/webhook/deployment_webhook.yaml:51]: This is needed to not break the test as cosigned fails to start up when unknown flag is configured and the `--webhook-name` flag will be available in the next release, which should be 1.8.0

Comment (hectorj2f) [reply charts/cosigned/templates/webhook/deployment_webhook.yaml:51]: Yes, I was thinking to pause this PR until we get a new release probably during the next week.

File: charts/cosigned/values.schema.json [hunk 5]
```diff
@@ -75,7 +76,8 @@
                     "secretKeyRef": {
                         "name": ""
                     },
-                    "cosignPub": ""
+                    "cosignPub": "",
+                    "webhookName": ""
```
Comment (hectorj2f) [inline charts/cosigned/values.schema.json:80]: Shouldn't it be `webhookName=cosigned.sigstore.dev` now ?

**Review by hectorj2f — COMMENTED** [review]
lgtm, i just added some minor comments. I am not sure if we should simply wait until the next release instead of adding an if statement for the version.

**Comment by hectorj2f** [comment]
Thanks for the PR @vpnachev :)

File: charts/cosigned/templates/webhook/deployment_webhook.yaml (hunk 4 above)
Comment (vpnachev) [reply charts/cosigned/templates/webhook/deployment_webhook.yaml:51]: Sounds like a plan to me. I can drop the commit with this change later.

Comment (hectorj2f) [reply charts/cosigned/templates/webhook/deployment_webhook.yaml:51]: cc @cpanato any thoughts ?

File: charts/cosigned/values.schema.json (hunk 5 above)
Comment (vpnachev) [reply charts/cosigned/values.schema.json]: Done.

Comment (vpnachev) [reply charts/cosigned/values.schema.json]: Probably, I will need to rebase after #128

**Comment by hectorj2f** [comment]
@vpnachev Please, rebase this PR when possible. Thanks.

**Comment by vpnachev** [comment]
The PR is rebased. Would you like me to contribute similar change for https://github.com/sigstore/cosign/pull/1757 in this PR or I should open new one? I am fine with both approaches.

**Comment by hectorj2f** [comment]
@vpnachev Yes, let's add a commit to this PR with a similar change for the other webhook. Thanks

**Comment by vpnachev** [comment]
The policy-webhook flags are now contributed. If this PR is hold until cosigned is updated to >=v1.8, I will drop the `{{- if semverCompare ">= 1.8-0" .Chart.AppVersion }}` checks, but for now I will keep them.

**Comment by hectorj2f** [comment]
@vpnachev lgtm for now :)

**Review by dlorenc — APPROVED** [review]

**Pull request merged**